### ISSUE TITLE
API aggregate requires update

### ISSUE THREAD

**Issue by kasperwelbers** [amcat/amcat#699]
I'm working on the R API bindings, and ran into some issues with api/v4/aggregate. This issue shamelessly combines a (what seems to be) bug and feature request.

**The project argument is ignored**
Aggregate via the API does not use the project argument, so if no sets are specified, all articles are used. For the following, results are identical, and include everything on the server (perhaps restricted by access):

http://vu.amcat.nl/api/v4/aggregate?q=*
http://vu.amcat.nl/api/v4/aggregate?q=*&project=7      

This might be a bit overkill, and it would be more useful to be able to search in all sets in a project. Also, it is inconsistent with 'search', which does take project into account. The following returns (1) an error and (2) only hits in project 7

http://vu.amcat.nl/api/v4/search?q=*
http://vu.amcat.nl/api/v4/search?q=*&project=7

**Limited axis options (compared to api/v4/query/aggregation)**

It's possible to aggregate by two axes, but the second axis can only be 'query', and the first axis doesn't accept text fields (including medium). 

https://github.com/amcat/amcat/blob/3074ab4d6a60afe5923b1a5e6c452253d5884cf1/api/rest/resources/aggregate.py#L37-L59

This is possible in api/v4/query/aggregation, so could the same backend be used here?

**Comment by vanatteveldt** [comment]
I disagree with using the project to filter sets, but I do think the set argument should be required. Agree?

**Comment by CJStuart** [comment]
Adding a project filter would complicate the behaviour of the aggregate view, so I agree that for simplicity's sake it's a good idea to let the user explicitly specify the required sets, rather than the project.

I don't know if it's useful to enforce a sets filter. ElasticSearch doesn't have any trouble aggregating on 18 million articles, so performance-wise it's not really necessary.
